### PR TITLE
Update 7715: Remove policies and update signer types

### PR DESCRIPTION
**Pull request by lukasrosario**
- remove policies. all constraints are specified in the `permissions` field.
- update signer types. specify secp256r1 or secp256k1 `key` types.

### CODE REVIEW

File: ERCS/erc-7715.md [hunk 1]
```diff
@@ -216,21 +198,23 @@ type WalletSigner = {
...
-// Each element of `ids` is a did:key identifier just like the `key` signer.
+// Each element of `data` is explicitly an secp256r1 (p256) or secp256k1 key, and the public keys are hex-encoded
 type MultiKeySigner = {
   type: "keys";
   data: {
-    ids: string[];
-  };
+    type: "secp256r1" | "secp256k1";
+    publicKey: `0x${string}`;
+  }[];
```
Comment (lukasrosario) [inline ERCS/erc-7715.md]: forgot exactly where we landed on this, does this seem right? @pedrouid @lukaisailovic

Comment (lukaisailovic) [reply ERCS/erc-7715.md]: @lukasrosario Each signer has `type` and `data` fields. We can't make `data` into array.
Proposing that we do this:
```
data: {
keys: {
type: ...
publicKey: ...
}[]
}
```

wdyt?

**Comment by lukaisailovic** [comment]
Since we reuse the same `type: "secp256r1" | "secp256k1";` should we extract it to `KeyType`? 
Also possible values should be:
- secp256k1
- secp256r1
- ed25519
- schnorr

File: ERCS/erc-7715.md (hunk 1 above)
Comment (lukasrosario) [reply ERCS/erc-7715.md]: this assumes all the keys would be the same type. are there use cases for mixing the types? if not this is fine with me

Comment (lukaisailovic) [reply ERCS/erc-7715.md]: @lukasrosario Our use case involves multiple keys with different types. Also all other signers have `data`, not `data[]`

Comment (lukasrosario) [reply ERCS/erc-7715.md]: how does what you wrote cover that use case then? there is only one `type`

Comment (lukasrosario) [reply ERCS/erc-7715.md]: i updated it, lmk if i misunderstood

Comment (lukaisailovic) [reply ERCS/erc-7715.md]: Can't add suggestion, pasting it here. Lets do this:
```
data: {
  keys: {
    type: KeyType;
    publicKey: `0x${string}`
  }[];
};
```

Comment (lukasrosario) [reply ERCS/erc-7715.md]: ah i see. misread your initial comment my bad

Comment (lukasrosario) [reply ERCS/erc-7715.md]: updated!

File: ERCS/erc-7715.md [hunk 2]
```diff
@@ -96,20 +90,14 @@ An array of `PermissionRequest` objects is the final `params` field expected by
...
+          {
             type: 'native-token-transfer',
             data: {
                 allowance: '0x1DCD6500'
             }
-        },
-        policies: [
-            {
-                type: 'gas-limit',
-                data: {
```
Comment (pedrouid) [inline ERCS/erc-7715.md:108]: Can you bring back the gas-limit in this example? So we could see it used in parallel with the transfer permission?

Comment (lukasrosario) [reply ERCS/erc-7715.md:108]: done!

File: ERCS/erc-7715.md [hunk 3]
```diff
@@ -312,7 +297,7 @@ type RateLimitPolicy = {
 };
 
 // The amount of Native Token that can be spent by a permission
-type NativeTokenSpendLimitPolicy = {
+type NativeTokenSpendLimitPermission = {
```
Comment (filmakarov) [inline ERCS/erc-7715.md]: Now, when it is a permission, what is the difference b/w this one and the first one, `NativeTokenTransferPermission` ?

Comment (lukasrosario) [reply ERCS/erc-7715.md]: good catch, just a duplicate in this case. removed

**Review by lukaisailovic — APPROVED** [review]
Lets 🚢 it!

**Comment by KannuSingh** [comment]
any reason we don't have `Contract Call` permissions type?

**Pull request merged**